docs/sdk-and-tools/sdk-nestjs/sdk-nestjs-monitoring.md

```diff
@@ -0,0 +1,197 @@
+
+const profiler = new PerformanceProfiler();
+await doSomething();
+const profilerDurationInMs = profiler.stop();
+
+console.log(`doSomething() method execution time lasted ${profilerDurationInMs} ms`);
+```
+
+The `.stop()` method can receive two optional parameters:
+- `description` - text used for default logging. Default: `undefined`
+- `log` - boolean to determine if log should be printed. If `log` is set to true, the logging class used to print will be `Logger` from `"@nestjs/common"`.` `Default: `false`
+
+```typescript
+import { PerformanceProfiler } from '@multiversx/sdk-nestjs-monitoring';
+
+const profiler = new PerformanceProfiler();
+await doSomething();
+profiler.stop(`doSomething() execution time`, true);
+```
+The output of the code above will be "`doSomething() execution time: 1.532ms`"
+
+---
+
+[comment]: # (mx-context-auto)
+
+### Cpu Profiler
+`CpuProfiler` is a class exported by the package that allows you to measure the CPU execution time of your code. Given that JavaScript is a single-threaded language, it's important to be mindful of the amount of CPU time allocated to certain operations, as excessive consumption can lead to slowdowns or even blockages in your process.
+
+```typescript
+import { CpuProfiler } from '@multiversx/sdk-nestjs-monitoring';
+
+const profiler = new CpuProfiler();
+await doHttpRequest()
+const profilerDurationInMs = profiler.stop();
+
+console.log(`doHttpRequest() method execution time lasted ${profilerDurationInMs} ms`);
+```
+
+The `.stop()` method can receive two optional parameters:
+- `description` - text used for default logging. Setting the description automatically triggers the printing of the `PerformanceProfiler` value. Default: `undefined`
+
+```typescript
+import { CpuProfiler } from '@multiversx/sdk-nestjs-monitoring';
+
+const httpReqCpuProfiler = new CpuProfiler();
+await doHttpRequest();
+httpReqCpuProfiler.stop(`doHttpRequest() execution time`);
+
+const cpuProfiler = new CpuProfiler();
+await doSomethingCpuIntensive();
+cpuProfiler.stop(`doSomethingCpuIntensive() execution time`);
+```
+The output of the code above will be <br/>
+
+`doHttpRequest() execution time: 100ms, CPU time: 1ms`
+`doSomethingCpuIntensive() execution time: 20ms, CPU time 18ms`
+
+*Note that a big execution time does not necessarily have an impact on the CPU load of the application. That means that, for example, while waiting for an HTTP request, the JavaScript thread can process other things. That is not the case for CPU time. When a method consumes a lot of CPU time, Javascript will not be able to process other tasks, potentially causing a freeze until the CPU-intensive task is complete.*
+
+---
+
+[comment]: # (mx-context-auto)
+
+## Interceptors
+The package provides a series of [Nestjs Interceptors](https://docs.nestjs.com/interceptors) which will automatically log and set the CPU and overall duration for each request in a prom histogram ready to be scrapped by Prometheus.
```

Comment:
`_prom_` or `Prometheus histogram`, plus a link to Prometheus?

Reply:
done